### PR TITLE
Add service principal in permissions doc

### PR DESCRIPTION
**Pull request by NastasiaSaby**
# What

Add a service principal in the doc example for Databricks permissions for jobs creation.

# Why

To get somewhere an example.

### CODE REVIEW

File: docs/resources/permissions.md [hunk 1]
```diff
@@ -151,6 +151,10 @@ resource "databricks_group" "eng" {
     display_name = "Engineering"
 }
 
+resource "databricks_service_principal" "main" {
+  display_name = "main"
```
Comment (nfx) [inline docs/resources/permissions.md:155]: Please also explicitly mention for which cloud is it. The API is rather confusing, unfortunately.

**Review by nfx — CHANGES_REQUESTED** [review]
Thank you for polishing our documentation.

Small comment to be added, as SPs have different definitions on AWS and Azure.

File: docs/resources/permissions.md (hunk 1 above)
Comment (NastasiaSaby) [reply docs/resources/permissions.md:155]: I did it by changing the name of the "databricks service principal".

Comment (nfx) [reply docs/resources/permissions.md:155]: Great!

**Pull request merged**